purchases/src/main/kotlin/com/revenuecat/purchases/Purchases.kt

```diff
@@ -1245,7 +1245,7 @@ class Purchases internal constructor(
                             }
                         } else {
                             storeProducts.firstOrNull() { product ->
-                                product.productId == purchase.productIds.firstOrNull()
+                                product.id == purchase.productIds.firstOrNull()
```

Comment:
I don't know if I love this now 😬

If though this logic is surrounded by an if statement making sure this isn't `ProductType.SUBS`, `product.id` isn't necessarily a 100% matchup of what would appear of `purchase.productIds`

☝️ This is because `product.Id` could contain `<product_id>:<base_plan_id>` where `purchase.productIds` would only contain `<product_id>`

I don't want this knowledge to have to be known since its `productl.id` and `purchase.productIds` seem like they would be equal here for Google's cause but its not

I'm going to think about this before taking this PR out of draft...